app/views/collections/_collect.html.erb

```diff
@@ -0,0 +1,9 @@
+<% if collect = current_user.collection_for(story) %>
```

Comment:
OK, I'll look into moving it to controller. The partial is called within the collection index view, where it searches the current user's collections and checks if they're empty or not. If empty, it displays a message stating there is nothing in collection. If it has something, it goes through the `current_user.collections.each do |collection|` logic and applies the icon change button.